### ISSUE TITLE
Radio Button (RadioButton) onPress Bug

### ISSUE THREAD

**Issue by elmcapp** [callstack/react-native-paper#2928]
There seem to be a bug with radio button onPress event. 

Does not work

`onPress={() => console.info('item._id')}

<RadioButton.Item
                    label={item.templateName}
                    value={item._ref}
                    onPress={() => console.info('item._id')}
                  />`

Will Work

`onPress={console.info('item._id')}

<RadioButton.Item
                    label={item.templateName}
                    value={item._ref}
                    onPress={console.info('item._id')}
                  />`

Notice that you can use ()=>

I am using  react-native-paper@4.9.2

**Comment by rengare** [comment]
Hi @elmcapp, you should use this code instead
``` tsx
const MyComponent = () => {
  const [value, setValue] = React.useState('first');

  const f = (newValue) => {
    setValue(newValue);
    alert('test')
  }

  return (
    <RadioButton.Group onValueChange={f} value={value}>
      <View>
        <Text>Second</Text>
        <RadioButton value="second" />
      </View>
    </RadioButton.Group>
  );
};

export default MyComponent
 ```
 If you really want to execute an action on single radio item then remove `RadioButtonGroup`

**Comment by elmcapp** [comment]
> Hi @elmcapp, you should use this code instead
> 
> ```tsx
> const MyComponent = () => {
>   const [value, setValue] = React.useState('first');
> 
>   const f = (newValue) => {
>     setValue(newValue);
>     alert('test')
>   }
> 
>   return (
>     <RadioButton.Group onValueChange={f} value={value}>
>       <View>
>         <Text>Second</Text>
>         <RadioButton value="second" />
>       </View>
>     </RadioButton.Group>
>   );
> };
> 
> export default MyComponent
> ```
> 
> If you really want to execute an action on single radio item then remove `RadioButtonGroup`

I tried this method. Where your <view> component is I had a flat list of RadioButton and the flattest was wrapped in the RadioButton.Group. I also tested your code but onPress event are not firing.

**Comment by elmcapp** [comment]
> Could you please specify what specifically doesn't work on `RadioButton`?

OnPress event will not fire

**Comment by kieran-osgood** [comment]
The onPress seems to only be called if onValueChanged hasn't been provided:
```
export const handlePress = ({
  onPress,
  value,
  onValueChange,
}: {
  onPress?: () => void;
  value: string;
  onValueChange?: (value: string) => void;
}) => {
  onValueChange ? onValueChange(value) : onPress?.();
};
```
from: https://github.com/callstack/react-native-paper/blob/main/src/components/RadioButton/utils.ts

I was about to submit a PR...but someone beat me to it! There is a PR to add a console warning here to explain to users mixing behaviours: https://github.com/callstack/react-native-paper/pull/2931 - 
so @elmcapp I updated your snack to assist in explaining: https://snack.expo.dev/l6EMGlAP4 essentially you have two options:
1. if you have wrapped your `<RadioButton.Item>`'s with `<RadioButton.Group>` - then the `onValueChanged` is the only event called callback
2. if you are using the `<RadioButton.Item>`'s outside of the `<RadioButton.Group>`'s, then the `onPress` will be called and the `onValueChanged` (if you were to pass that directly into the `Item`) will not (see <OptionTwo /> in the snack I created)
